keras_cv/bounding_box/iou.py

```diff
@@ -50,16 +95,19 @@ def compute_iou(
     boxes1_rank = len(boxes1.shape)
     boxes2_rank = len(boxes2.shape)
 
-    if (
-        boxes1_rank != boxes2_rank
-        or boxes1_rank not in [2, 3]
-        or boxes2_rank not in [2, 3]
-    ):
+    if boxes1_rank not in [2, 3]:
```

Comment:
can we update the docstring to add more information about the updated input/output shapes?

Reply:
Done.

Reply:
I have added 4 different scenarios on the documentation part so hopefully this will give user more context